connman: 1.41 -> 1.42

eclairevoyant opened the pull request:
## Description of changes

Bump version, remove patches that are part of the release, and use SRI hashes

Also replaced maintainer with myself as it seems they are part of the retired contributors group

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

eclairevoyant committed:
connman: 1.41 -> 1.42

lf- commented:
btw your commit message has the wrong versions, it's 1.41 and 1.42 :)

bjornfor commented:
> Bump version, use finalAttrs idiom instead of rec, remove patches that are part of the release, and use SRI hashes
>
> Also replaced maintainer with myself as it seems they are part of the retired contributors group

This sounds like the 1 commit should be split into 3:
* bump
* use finalAttrs pattern
* update maintainers

eclairevoyant committed:
connman: update maintainers

remove matejc as they are part of the team https://github.com/orgs/NixOS/teams/retired-nixpkgs-contributors

eclairevoyant commented:
@lf- Thanks for catching, fixed.
@bjornfor After discussing elsewhere further, I removed the `finalAttrs` change as anyone overriding would need to provide the fetcher for `src` anyway (due to changed hash), therefore `finalAttrs` provides no benefit here in terms of overriding. Remaining changes have been split as requested.

eclairevoyant commented:
@lf- @bjornfor are there any other blockers to merging this?

bjornfor commented:
I don't have any objections, but I don't feel particularly qualified to approve/merge this either.

> connman: update maintainers 

Can you write in the commit message why you're removing the existing maintainer?

Does the patch not have an upstream we can fetch from? Will it be upstreamed? These are things worth writing a comment about in the code or commit message.

lf- commented:
> Can you write in the commit message why you're removing the existing maintainer?
To my understanding it was because the previous maintainer is not working on the package for several versions.

> Does the patch not have an upstream we can fetch from? Will it be upstreamed? These are things worth writing a comment about in the code or commit message.

Upstream hecked up their dist tarball for 1.42. The fix was committed a few days ago so it will be fixed in the next release, we can just delete it after. The file is simply taken from the repo: #250307 is another approach to the same thing, by not using a dist tarball, instead just using the git tag.

eclairevoyant commented:
@bjornfor I basically reused part of one of the previous patches because of the broken dist tarball, added comment for it. maintainer-related commit has also been updated as requested.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2633

wegank commented on pkgs/tools/networking/connman/connman/default.nix:
Is it possible to replace it with fetchpatch?

eclairevoyant replied:
Unfortunately it's only 1/3 of the diff associated with that commit, the other files already have the patch applied

alyssais replied:
For next time, it's possible to tell fetchpatch to exclude/include specific files.

eclairevoyant replied:
didn't realise that, thanks.

alyssais commented on pkgs/tools/networking/connman/connman/default.nix:
This hash is just not correct:

```
% nix --extra-experimental-features nix-command hash to-sri --type sha256 1kg2nml7pdxc82h5hgsa3npvzdxy4d2jpz2f93pa97if868i8d43
sha256-gzQUkUEunqTuSE78K0Ujvre/rx1KP1igQKy3e2i14s0=
```

Fix in https://github.com/NixOS/nixpkgs/pull/255103.
